syz-verifier/verifier.go

```diff
@@ -1,397 +1,617 @@
+					if level != 0 {
+						return
+					}
+					log.Logf(level, msg, args...)
+				},
+				NewInputFilter: func(call string) bool {
+					vrf.mu.Lock()
+					defer vrf.mu.Unlock()
+					return !vrf.saturatedCalls[call]
+				},
+			}, rnd, vrf.target)
+			fuzzerObj.AddCandidates(candidates)
+			vrf.fuzzer.Store(fuzzerObj)
+			vrf.http.Fuzzer.Store(fuzzerObj)
+			go vrf.corpusInputHandler(corpusUpdates)
+			go vrf.corpusMinimization()
+			vrf.phase = vrfPhaseFuzzing
+			log.Logf(0, "fuzzer started with %d candidates", len(candidates))
+		}
+		if vrf.phase == vrfPhaseFuzzing {
+			req := vrf.fuzzer.Load().Next()
+			if req == nil {
+				log.Logf(0, "no more candidates to fuzz, waiting for new candidates")
+				// Wait for new candidates to be added.
+				select {
+				case <-ctx.Done():
+					return
+				case <-time.After(time.Second):
+					continue
+				}
+			}
+
+			// Distribute the same request to all kernel queues
+			var wg sync.WaitGroup
+			wg.Add(len(vrf.sources))
+			distributed := 0
+			log.Logf(3, "distributing program: %s to %d kernels", req.Prog.String(), len(vrf.sources))
+			for kernelID, source := range vrf.sources {
+				log.Logf(3, "distributing program to kernel %d: %s", kernelID, req.Prog.String())
+				reqCopy := &queue.Request{
+					Type:            req.Type,
+					ExecOpts:        req.ExecOpts,
+					Prog:            req.Prog,
+					BinaryFile:      req.BinaryFile,
+					GlobPattern:     req.GlobPattern,
+					ReturnAllSignal: req.ReturnAllSignal,
+					ReturnError:     req.ReturnError,
+					ReturnOutput:    req.ReturnOutput,
+					Stat:            req.Stat,
+					Important:       req.Important,
+					Avoid:           req.Avoid,
+				}
+				reqCopy.OnDone(func(req *queue.Request, res *queue.Result) bool {
+					log.Logf(3, "got result for kernel:%d %s: %v", kernelID, reqCopy.Prog.String(), res)
+					wg.Done()
+					return true
+				})
+				source.Submit(reqCopy)
+				distributed++
+			}
+			log.Logf(2, "distributed program to %d kernels", distributed)
+			wg.Wait()
+			log.Logf(3, "all %d kernels finished execution", len(vrf.sources))
+			log.Logf(3, "comparing results for %d kernels", len(vrf.sources))
+			// TODO: Implement result comparison logic to detect behavioral differences
+
+		}
 	}
-	defer inst.Close()
-	defer vrf.srv.cleanup(poolID, vmID)
+}
 
-	fwdAddr, err := inst.Forward(vrf.srv.port)
+// =============================================================================
+// Kernel
+// =============================================================================
+
+func (kernel *Kernel) FuzzerInstance(ctx context.Context, inst *vm.Instance, updInfo dispatcher.UpdateInfo) {
+	index := inst.Index()
+	injectExec := make(chan bool, 10)
+	kernel.serv.CreateInstance(index, injectExec, updInfo)
+	rep, err := kernel.runInstance(ctx, inst, injectExec)
+	lastExec, _ := kernel.serv.ShutdownInstance(index, rep != nil)
+	if rep != nil {
+		rpcserver.PrependExecuting(rep, lastExec)
+		select {
+		case kernel.crashes <- rep:
```

Comment:
It should be enough to just log it and ignore - `syz-verifier` is not about these kernel crashes.